packages/shared/lib/core/router/subrouters/profile-recovery-router.ts

```diff
@@ -0,0 +1,71 @@
+import { get, writable } from 'svelte/store'
+
+import { isGettingMigrationData, onboardingProfile, ProfileRecoveryType } from '@contexts/onboarding'
+
+import { onboardingRouter } from '../onboarding-router'
+import { ProfileRecoveryRoute } from '../enums'
+import { Subrouter } from './subrouter'
+
+export const profileRecoveryRoute = writable<ProfileRecoveryRoute>(null)
+export const profileRecoveryRouter = writable<ProfileRecoveryRouter>(null)
+
+export class ProfileRecoveryRouter extends Subrouter<ProfileRecoveryRoute> {
+    public importFile: Buffer
+
+    constructor() {
+        super(getInitialRoute() ?? ProfileRecoveryRoute.TextImport, profileRecoveryRoute, get(onboardingRouter))
+    }
+
+    resetRoute(): void {
+        profileRecoveryRoute.set(getInitialRoute() ?? ProfileRecoveryRoute.TextImport)
+    }
+
+    next(): void {
+        let nextRoute: ProfileRecoveryRoute
+
+        const currentRoute = get(this.routeStore)
+        switch (currentRoute) {
+            case ProfileRecoveryRoute.TextImport: {
+                const _profileRecoveryType = get(onboardingProfile)?.recoveryType
+                if (_profileRecoveryType === ProfileRecoveryType.Seed) {
+                    isGettingMigrationData.set(true)
+                    // await getMigrationData(migrationSeed)
+                    isGettingMigrationData.set(false)
+                    this.parentRouter.next()
```

Comment:
Yeah, that sounds good :+1:

Reply:
I will remove it in this one place, but we should create a dedicate PR to remove it entirely from the app. Most of it is in onbarding, but 1) there's some Ledger stuff that I won't want to mess with too much since Umair is working on it and 2) there's a Migrate Ledger index setting in the advanced profile settings for Ledger profiles.